### PR TITLE
[FAT-138]Implement API Karate tests

### PR DESCRIPTION
**Pull request by KyryloShulzhenko**
Implement undefined scenarios for mod-source-record-storage feature

Purpose/Overview:

Karate implementation based on test plans is created

Acceptance criteria:

API Tests are created at  https://github.com/folio-org/folio-integration-tests
Test are shown as passed in Cucumber report.

### CODE REVIEW

**Review by RomanChernetskyi — APPROVED** [review]

**Review by KaterynaSenchenko — APPROVED** [review]

File: mod-source-record-storage/src/main/resources/folijet/mod-source-record-storage/features/get-source-storage-records.feature [hunk 1]
```diff
@@ -5,51 +5,410 @@ Feature: Source-Record-Storage
...
+    Given path 'source-storage','records'
+    And request record4
+    When method POST
+    Then status 201
+    #   Get records
+    Given path 'source-storage', 'records'
+    And param recordType = "MARC_HOLDING"
+    When method GET
+    Then status 200
+    And print 'hello',response
```
Comment (RuslanLavrov) [inline mod-source-record-storage/src/main/resources/folijet/mod-source-record-storage/features/get-source-storage-records.feature:232]: please, remove this debug message

File: mod-source-record-storage/src/main/resources/folijet/mod-source-record-storage/features/get-source-storage-records.feature [hunk 2]
```diff
@@ -5,51 +5,410 @@ Feature: Source-Record-Storage
...
+    When method POST
+    Then status 201
+    #   Create record 3
+    Given path 'source-storage','records'
+    And request record3
+    When method POST
+    Then status 201
+    #   Get collections of records filtered by snapshot
+    Given path 'source-storage', 'records'
+    And param snapshotId = snapshot1.id
```
Comment (RuslanLavrov) [inline mod-source-record-storage/src/main/resources/folijet/mod-source-record-storage/features/get-source-storage-records.feature:61]:
```suggestion
    And param snapshotId = snapshot1.jobExecutionId
```

File: mod-source-record-storage/src/main/resources/folijet/mod-source-record-storage/features/get-source-storage-records.feature [hunk 3]
```diff
@@ -5,51 +5,410 @@ Feature: Source-Record-Storage
...
+    When method POST
+    Then status 201
+    #   Get collection filtered by snapshot and ordered, verify correct order
+    Given path 'source-storage', 'records'
+    And param snapshotId = snapshotId
+    And param recordType = 'EDIFACT'
+    And param orderBy = 'order,ASC'
+    When method GET
+    Then status 200
+    And assert response.records[2].order == 4
```
[the remark this answers is not in the document]
Comment (KyryloShulzhenko) [reply mod-source-record-storage/src/main/resources/folijet/mod-source-record-storage/features/get-source-storage-records.feature]: done

File: mod-source-record-storage/src/main/resources/folijet/mod-source-record-storage/features/get-source-storage-records.feature (hunk 1 above)
Comment (KyryloShulzhenko) [reply mod-source-record-storage/src/main/resources/folijet/mod-source-record-storage/features/get-source-storage-records.feature]: done

File: mod-source-record-storage/src/main/resources/folijet/mod-source-record-storage/features/get-source-storage-records.feature (hunk 2 above)
Comment (KyryloShulzhenko) [reply mod-source-record-storage/src/main/resources/folijet/mod-source-record-storage/features/get-source-storage-records.feature]: done

File: mod-source-record-storage/src/main/resources/folijet/mod-source-record-storage/features/get-source-storage-records.feature [hunk 4]
```diff
@@ -5,51 +5,410 @@ Feature: Source-Record-Storage
...
+    #   Create record 3
+    Given path 'source-storage','records'
+    And request record3
+    When method POST
+    Then status 201
+    #   Get collections of records filtered by snapshot
+    Given path 'source-storage', 'records'
+    And param snapshotId = snapshot1.id
+    When method GET
+    Then status 200
```
[the remark this answers is not in the document]
Comment (KyryloShulzhenko) [reply mod-source-record-storage/src/main/resources/folijet/mod-source-record-storage/features/get-source-storage-records.feature:63]: Done

File: mod-source-record-storage/src/main/resources/folijet/mod-source-record-storage/features/get-source-storage-records.feature [hunk 5]
```diff
@@ -5,51 +5,411 @@ Feature: Source-Record-Storage
...
+    #   Create record 1
+    Given path 'source-storage','records'
+    And request record1
+    When method POST
+    Then status 201
+    # Get record by id
+    Given path 'source-storage', 'records', recordId
+    When method GET
+    Then status 200
+    And assert response.id == recordId
```
Comment (RuslanLavrov) [inline mod-source-record-storage/src/main/resources/folijet/mod-source-record-storage/features/get-source-storage-records.feature:257]: please, add verification, according to test description, that error record is created
```suggestion
    And assert response.id == recordId
    And match response.errorRecord == '#notnull'
```
please, add verification, according to test description, that error record is created

Comment (KyryloShulzhenko) [reply mod-source-record-storage/src/main/resources/folijet/mod-source-record-storage/features/get-source-storage-records.feature:257]: done

**Review by RuslanLavrov — APPROVED** [review]

**Pull request merged**